src/inspect_ai/_util/constants.py

```diff
@@ -44,4 +44,8 @@
 DEFAULT_BATCH_SIZE = 100
 
 DESERIALIZING = "deserializing"
-DESERIALIZING_CONTEXT = {DESERIALIZING: True}
+MESSAGE_CACHE = "message_cache"
+
+
+def get_deserializing_context() -> dict[str, Any]:
+    return {DESERIALIZING: True, MESSAGE_CACHE: {}}
```

Comment:
This returns a fresh cache with each invocation, which seems like a potential footgun (honestly, though, so does the opposite)

Reply:
Yeah, I think it's fine and probably in fact best to bound the cache to the current deserialization context (which is a very top level thing)